lib/faktory/testing.rb

```diff
@@ -0,0 +1,329 @@
+require 'faktory/testing/runner'
+
+module Faktory
+  module Testing
+    def self.__test_mode
+      @__test_mode
+    end
+
+    def self.__test_mode=(mode)
+      @__test_mode = mode
+    end
+
+    def self.__set_test_mode(mode)
+      if block_given?
+        current_mode = self.__test_mode
+        begin
+          self.__test_mode = mode
+          yield
+        ensure
+          self.__test_mode = current_mode
+        end
+      else
+        self.__test_mode = mode
+      end
+    end
+
+    def self.disable!(&block)
+      __set_test_mode(:disable, &block)
+    end
+
+    def self.fake!(&block)
+      __set_test_mode(:fake, &block)
+    end
+
+    def self.inline!(&block)
+      # Don't allow blockless inline
+      # https://github.com/mperham/sidekiq/issues/3495
+      unless block_given?
+        raise 'Must provide a block to Faktory::Testing.inline!'
+      end
+
+      __set_test_mode(:inline, &block)
+    end
+
+    def self.enabled?
+      self.__test_mode != :disable
+    end
+
+    def self.disabled?
+      self.__test_mode == :disable
+    end
+
+    def self.fake?
+      self.__test_mode == :fake
+    end
+
+    def self.inline?
+      self.__test_mode == :inline
+    end
+
+    def self.constantize(str)
```

Comment:
Need to pick up the latest impl here:

https://github.com/mperham/sidekiq/blob/master/lib/sidekiq/processor.rb#L216